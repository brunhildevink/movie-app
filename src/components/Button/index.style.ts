import styled from 'styled-components'

const Button = styled.button`
  width: 32px;
  background: transparent;
  cursor: pointer;
  border: none;
  padding: 0;

  &:disabled {
    opacity: 0.5;
  }
`

export { Button }
