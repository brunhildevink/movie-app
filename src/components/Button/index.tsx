import React from 'react'
import { Button } from './index.style'
import { colors } from '../../styles'
import Icon from '../Icon'

interface Props {
  id?: string
}

const Prev: React.FC<Props> = ({ id }) => (
  <Button id={id}>
    <Icon.ArrowLeft fill={colors.white} display="inline" height={24} width={32} title="previous" />
  </Button>
)

Prev.defaultProps = {
  id: '',
}

const Next: React.FC<Props> = ({ id }) => (
  <Button id={id}>
    <Icon.ArrowRight fill={colors.white} display="inline" height={24} width={32} title="previous" />
  </Button>
)

Next.defaultProps = {
  id: '',
}

export default { Prev, Next }
