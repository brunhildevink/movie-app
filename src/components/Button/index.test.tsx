import React from 'react'
import renderer from 'react-test-renderer'
import Button from '.'

test('Button Prev renders correctly', () => {
  const tree = renderer.create(<Button.Prev />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Button Next renders correctly', () => {
  const tree = renderer.create(<Button.Next />).toJSON()
  expect(tree).toMatchSnapshot()
})
