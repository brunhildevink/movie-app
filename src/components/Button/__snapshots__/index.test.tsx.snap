// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Next renders correctly 1`] = `
<button
  className="sc-bczRLJ epWIDt"
  id=""
>
  <svg
    style={
      Object {
        "display": "inline",
        "fill": "#ffffff",
        "height": 24,
        "lineHeight": 1,
        "verticalAlign": "middle",
        "width": 32,
      }
    }
    viewBox="0 0 256 256"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      previous
    </title>
    <path
      d="M0 0h256v256H0z"
      fill="none"
    />
    <path
      d="M40 128h176M144 56l72 72-72 72"
      fill="none"
      stroke="#ffffff"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
  </svg>
</button>
`;

exports[`Button Prev renders correctly 1`] = `
<button
  className="sc-bczRLJ epWIDt"
  id=""
>
  <svg
    style={
      Object {
        "display": "inline",
        "fill": "#ffffff",
        "height": 24,
        "lineHeight": 1,
        "verticalAlign": "middle",
        "width": 32,
      }
    }
    viewBox="0 0 256 256"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      previous
    </title>
    <path
      d="M0 0h256v256H0z"
      fill="none"
    />
    <path
      d="M216 128H40M112 56l-72 72 72 72"
      fill="none"
      stroke="#ffffff"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
  </svg>
</button>
`;
